apps/web/components/banner.tsx

```diff
@@ -26,17 +26,47 @@ export const Banner: React.FC<PropsWithChildren<Props>> = ({
   variant,
   persistChoice,
 }) => {
-  const [visible, setVisible] = persistChoice
+  const bannerHeightRef = useRef<number>(0);
+  const bannerRef = useRef<HTMLDivElement>(null);
+
+  const [visible, setVisible] = !persistChoice
     ? useState(true)
-    : useLocalStorage(`unkey_banner_${persistChoice}`, true);
+    : useLocalStorage(`unkey_banner_${persistChoice}`, true, { initializeWithValue: false });
+
+  useResizeObserver({
+    ref: bannerRef,
+    onResize: ({ height }) => {
+      if (height !== bannerHeightRef.current) {
+        // Update Banners.
+        updateBannersBottomPosition();
+
+        // Set current banner height.
+        bannerHeightRef.current = height ?? 0;
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (visible && bannerRef.current) {
+      updateBannersBottomPosition();
+
+      bannerHeightRef.current = bannerRef.current.getBoundingClientRect().height;
+    }
+  }, [visible]);
+
   if (!visible) {
     return null;
   }
+
   return (
-    <div className="fixed inset-x-0 bottom-0 z-50 pointer-events-none sm:flex sm:justify-center sm:px-6 sm:pb-5 lg:px-8">
+    <div
+      ref={bannerRef}
+      style={{ opacity: 0 }}
+      className="Unkey__Banner fixed inset-x-0 bottom-0 z-50 pointer-events-none sm:flex sm:justify-center sm:px-6 sm:mb-5 lg:px-8"
```

Comment:
Necessary "sm:mb-5" instead of "sm:pb-5" so that the height is calculated without the spacing. Otherwise, complicates the calculation since it's a responsive class,